engine/src/test/java/org/camunda/bpm/engine/test/api/mgmt/telemetry/TelemetryReporterTest.java

```diff
@@ -315,6 +357,76 @@ public void shouldReportInitialDataWhenReporterActivatedAndInitTelemetryEnabled(
+    // given
+    ProcessEngineConfigurationImpl processEngineConfiguration = createEngineWithInitMessage(null);
+    stubFor(post(urlEqualTo(TELEMETRY_ENDPOINT_PATH))
+            .willReturn(aResponse()
+                        .withStatus(HttpURLConnection.HTTP_ACCEPTED)));
+
+    Data expectedData = createInitialDataToSend(processEngineConfiguration.getTelemetryData(), null);
+    String requestBody = new Gson().toJson(expectedData);
+
+    // when
+    processEngineConfiguration.getTelemetryReporter().reportNow();
+    standaloneProcessEngine.close();
+    standaloneProcessEngine = null;
+
+    // then
+    verify(1, postRequestedFor(urlEqualTo(TELEMETRY_ENDPOINT_PATH))
+              .withRequestBody(equalToJson(requestBody, JSONCompareMode.LENIENT))
+              .withHeader("Content-Type",  equalTo("application/json")));
+  }
+
+
+  @Test
+  public void shouldReportInitialDataOnceInitTelemetryDisabled() {
+    // given
+    ProcessEngineConfigurationImpl processEngineConfiguration = createEngineWithInitMessage(false);
+    stubFor(post(urlEqualTo(TELEMETRY_ENDPOINT_PATH))
+            .willReturn(aResponse()
+                        .withStatus(HttpURLConnection.HTTP_ACCEPTED)));
+
+    Data expectedData = createInitialDataToSend(processEngineConfiguration.getTelemetryData(), false);
+    String requestBody = new Gson().toJson(expectedData);
+
+    // when
+    processEngineConfiguration.getTelemetryReporter().reportNow();
+    standaloneProcessEngine.close();
+    standaloneProcessEngine = null;
+
+    // then
+    verify(1, postRequestedFor(urlEqualTo(TELEMETRY_ENDPOINT_PATH))
+              .withRequestBody(equalToJson(requestBody, JSONCompareMode.LENIENT))
+              .withHeader("Content-Type",  equalTo("application/json")));
+  }
+
+  @Test
+  @WatchLogger(loggerNames = {"org.camunda.bpm.engine.telemetry"}, level = "DEBUG")
+  public void shouldReportInitialDataOnceInitTelemetryEnabled() {
+    // given
+    ProcessEngineConfigurationImpl processEngineConfiguration = createEngineWithInitMessage(true);
+    stubFor(post(urlEqualTo(TELEMETRY_ENDPOINT_PATH))
+            .willReturn(aResponse()
+                        .withStatus(HttpURLConnection.HTTP_ACCEPTED)));
+
+    Data expectedData = createInitialDataToSend(processEngineConfiguration.getTelemetryData(), true);
+    String requestBody = new Gson().toJson(expectedData);
+
+    // when
+    processEngineConfiguration.getTelemetryReporter().reportNow();
+    standaloneProcessEngine.close();
+    standaloneProcessEngine = null;
+
+    // then
+    verify(3, postRequestedFor(urlEqualTo(TELEMETRY_ENDPOINT_PATH))
+              .withRequestBody(equalToJson(requestBody, JSONCompareMode.LENIENT))
+              .withHeader("Content-Type",  equalTo("application/json")));
+    assertThat(loggingRule.getFilteredLog("Sending initial telemetry data").size()).isOne();
+    assertThat(loggingRule.getFilteredLog("Initial telemetry request was successful.").size()).isOne();
+  }
```

Comment:
Do we need new test cases for this or should we simply integrate this into the other test cases directly above (in the end, they simply make it more specific that initial data should only be sent once)?

Reply:
Now I spotted the difference! Cool, let's keep them 👍